libraries/abstractions/transport/utest/transport_secure_sockets_utest.c

```diff
@@ -0,0 +1,740 @@
+
+/**
+ * @brief Test #SecureSocketsTransport_Disconnect with invalid parameters.
+ */
+void test_SecureSocketsTransport_Disconnect_NULL_Network_Context( void )
+{
+    TransportSocketStatus_t returnStatus;
+
+    returnStatus = SecureSocketsTransport_Disconnect( NULL );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INVALID_PARAMETER, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test the #SecureSocketsTransport_Disconnect and encounterd #SOCKETS_Shutdown failure.
+ */
+void test_SecureSocketsTransport_Disconnect_Fail_to_ShutDown( void )
+{
+    TransportSocketStatus_t returnStatus;
+
+    SOCKETS_Shutdown_ExpectAnyArgsAndReturn( MOC_SECURE_SOCKET_ERROR );
+    returnStatus = SecureSocketsTransport_Disconnect( &networkContext );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test #SecureSocketsTransport_Disconnect and encounterd #SOCKETS_Close failure.
+ */
+void test_SecureSocketsTransport_Disconnect_Fail_to_Close( void )
+{
+    TransportSocketStatus_t returnStatus;
+
+    SOCKETS_Shutdown_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    SOCKETS_Close_ExpectAnyArgsAndReturn( MOC_SECURE_SOCKET_ERROR );
+    returnStatus = SecureSocketsTransport_Disconnect( &networkContext );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_INTERNAL_ERROR, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Happy path #SecureSocketsTransport_Disconnect.
+ */
+void test_SecureSocketsTransport_Disconnect_Sucess( void )
+{
+    TransportSocketStatus_t returnStatus;
+
+    SOCKETS_Shutdown_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    SOCKETS_Close_ExpectAnyArgsAndReturn( SOCKETS_ERROR_NONE );
+    returnStatus = SecureSocketsTransport_Disconnect( &networkContext );
+    TEST_ASSERT_EQUAL( TRANSPORT_SOCKET_STATUS_SUCCESS, returnStatus );
+}
+
+/*-----------------------------------------------------------*/
+
+/**
+ * @brief Test that #SecureSocketsTransport_Send is able to return that 0 bytes are sent over
+ * the network stack when passing any invalid parameters.
+ */
+void test_SecureSocketsTransport_Send_Send_Invalid_Params( void )
+{
+    int32_t bytesSent;
+    NetworkContext_t invalidNetworkContext = { 0 };
+    invalidNetworkContext.tcpSocket = NULL;
+
+    bytesSent = SecureSocketsTransport_Send( NULL, networkBuffer, BYTES_TO_SEND );
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesSent );
+
+    bytesSent = SecureSocketsTransport_Send( &invalidNetworkContext, NULL, BYTES_TO_SEND );
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesSent );
+
+    bytesSent = SecureSocketsTransport_Send( &invalidNetworkContext, networkBuffer, 0 );
+    TEST_ASSERT_EQUAL( SOCKETS_EINVAL, bytesSent );
+
+    bytesSent = SecureSocketsTransport_Send( &invalidNetworkContext, networkBuffer, BYTES_TO_SEND );
```

Comment:
Can you explicitly set invalidNetworkContext to have a SOCKETS_INVALID_SOCKET above this line, like you did in test_SecureSocketsTransport_Connect_Invalid_Params()?

```suggestion
    invalidNetworkContext.tcpSocket = SOCKETS_INVALID_SOCKET;
    bytesSent = SecureSocketsTransport_Send( &invalidNetworkContext, networkBuffer, BYTES_TO_SEND );
```